scripts/bulk_update_leads_info.py

```diff
@@ -0,0 +1,105 @@
+#-*- coding: utf-8 -*-
+
+import re
+import argparse
+import csv
+import logging
+from closeio_api import Client as CloseIO_API, APIError
+
+get_contact_info = lambda key, row, what, typ: [{what: row[x], 'type': typ} for x in row.keys()
+                                                if re.match(r'contact%s_%s[0-9]' % (key, what), x) and row[x]]
+
+parser = argparse.ArgumentParser(description='')
+parser.add_argument('csvfile', type=argparse.FileType('rU'), help='csv file')
+parser.add_argument('--api_key', '-k', required=True, help='API Key')
+parser.add_argument('--development', '-d', action='store_true',
+                    help='Use a development (testing) server rather than production.')
+parser.add_argument('--confirmed', '-c', action='store_true',
+                    help='Without this flag, the script will do a dry run without actually updating any data.')
+parser.add_argument('--disable-create', '-x', action='store_true',
+                    help='Prevent new lead creation. Update only exists leads.')
+args = parser.parse_args()
+
+log_format = "[%(asctime)s] %(levelname)s [%(name)s.%(funcName)s:%(lineno)d] %(message)s"
+if not args.confirmed:
+    log_format = 'DRY RUN: '+log_format
+logging.basicConfig(level=logging.INFO, format=log_format)
+logging.debug('parameters: %s' % vars(args))
+
+sniffer = csv.Sniffer()
+dialect = sniffer.sniff(args.csvfile.read(1024))
+args.csvfile.seek(0)
+c = csv.DictReader(args.csvfile, dialect=dialect)
+assert any(x in ('company', 'lead_id') for x in c.fieldnames), \
+    'ERROR: column "company" or "lead_id" is not found'
+
+
+api = CloseIO_API(args.api_key, development=args.development)
+
+try:
+    resp = api.get('custom_fields/lead/')
+except APIError as e:
+    logging.error('line: %d : %s' % (c.line_num, e))
+    raise
+
+available_custom_fieldnames = [x['name'] for x in resp['data']]
+new_custom_fieldnames = [x for x in [y.split('.')[1] for y in c.fieldnames if y.startswith('custom.')]
+                         if x not in available_custom_fieldnames]
+
+for field in new_custom_fieldnames:
```

Comment:
let's only create new custom fields if `--create-custom-fields` is passed - that way it makes it obvious if there are typos in the column names, and gives people a chance to setup the proper "type" (e.g. Number) of custom field ahead of time if they want